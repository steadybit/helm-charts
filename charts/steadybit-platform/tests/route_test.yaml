templates:
  - route.yaml
chart:
  appVersion: 0.0.0
  version: 0.0.0
tests:
  - it: should disable route
    set:
      route:
        enabled: false
    asserts:
      - hasDocuments:
          count: 0
  - it: should enable route
    set:
      route:
        enabled: true
        subdomain: example
    asserts:
      - matchSnapshot: { }
  - it: should disable route in plain kubernetes
    asserts:
      - hasDocuments:
          count: 0
  - it: should enable route in openshift
    set:
      route:
        host: example.steadybit.com
        annotations:
          this: that
        tls:
          termination: edge
          insecureEdgeTerminationPolicy: Redirect
    capabilities:
      apiVersions:
        - route.openshift.io/v1
    asserts:
      - matchSnapshot: { }
