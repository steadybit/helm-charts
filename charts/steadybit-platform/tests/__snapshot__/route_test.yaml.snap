should enable route:
  1: |
    apiVersion: route.openshift.io/v1
    kind: Route
    metadata:
      name: RELEASE-NAME-steadybit-platform
      namespace: NAMESPACE
    spec:
      host: null
      path: /
      port:
        targetPort: ui-port
      subdomain: example
      tls:
        insecureEdgeTerminationPolicy: Redirect
        termination: edge
      to:
        kind: Service
        name: RELEASE-NAME-steadybit-platform
  2: |
    apiVersion: route.openshift.io/v1
    kind: Route
    metadata:
      name: RELEASE-NAME-steadybit-platform-ws
      namespace: NAMESPACE
    spec:
      host: null
      path: /ws
      port:
        targetPort: websocket-port
      subdomain: example
      tls:
        insecureEdgeTerminationPolicy: Redirect
        termination: edge
      to:
        kind: Service
        name: RELEASE-NAME-steadybit-platform
should enable route in openshift:
  1: |
    apiVersion: route.openshift.io/v1
    kind: Route
    metadata:
      annotations:
        this: that
      name: RELEASE-NAME-steadybit-platform
      namespace: NAMESPACE
    spec:
      host: example.steadybit.com
      path: /
      port:
        targetPort: ui-port
      subdomain: null
      tls:
        insecureEdgeTerminationPolicy: Redirect
        termination: edge
      to:
        kind: Service
        name: RELEASE-NAME-steadybit-platform
  2: |
    apiVersion: route.openshift.io/v1
    kind: Route
    metadata:
      annotations:
        this: that
      name: RELEASE-NAME-steadybit-platform-ws
      namespace: NAMESPACE
    spec:
      host: example.steadybit.com
      path: /ws
      port:
        targetPort: websocket-port
      subdomain: null
      tls:
        insecureEdgeTerminationPolicy: Redirect
        termination: edge
      to:
        kind: Service
        name: RELEASE-NAME-steadybit-platform
